Sources/Sentry/include/SentryTransport.h

```diff
@@ -6,14 +6,25 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+typedef NS_ENUM(NSInteger, SentryFlushResult) {
+    kSentryFlushResultSuccess = 0,
+    kSentryFlushResultTimedOut,
+    kSentryFlushResultAlreadyFlushing,
+};
+
 NS_SWIFT_NAME(Transport)
 @protocol SentryTransport <NSObject>
 
 - (void)sendEnvelope:(SentryEnvelope *)envelope NS_SWIFT_NAME(send(envelope:));
 
 - (void)recordLostEvent:(SentryDataCategory)category reason:(SentryDiscardReason)reason;
 
-- (BOOL)flush:(NSTimeInterval)timeout;
+- (SentryFlushResult)flush:(NSTimeInterval)timeout;
+
+/**
+ * Only needed for testing.
+ */
+- (void)setStartFlushCallback:(void (^)(void))callback;
```

Comment:
m: I believe that everything that is meant only for test should be wrapper by `#if TEST` to save a few bytes, this applies to implementation as well.

Also, this function is only required for tests with `SentryHTTPTransport`. Therefore, it could be placed in the `SentryHTTPTransport` class, eliminating the need to add an empty function for TestTransport.